docs/src/main/asciidoc/kafka.adoc

```diff
@@ -3,438 +3,946 @@ This guide is maintained in the main Quarkus repository
-The data is sent to a browser using server-sent events.
+And all this functionality is provided in a distributed, highly scalable, elastic, fault-tolerant, and secure manner.
 
-image::kafka-guide-architecture.png[alt=Architecture, align=center]
+== Quarkus extension for Apache Kafka
 
-== Solution
+Quarkus provides support for Apache Kafka through https://smallrye.io/smallrye-reactive-messaging/[Smallrye Reactive Messaging] framework. 
+Based on Eclipse MicroProfile Reactive Messaging specification 2.0, it proposes a flexible programming model bridging CDI and event-driven.
 
-We recommend that you follow the instructions in the next sections and create the application step by step.
-However, you can go right to the completed example.
+You can add the `smallrye-reactive-messaging-kafka` extensions to your project by running the following command in your project base directory:
 
-Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+[source,bash]
+----
+./mvnw quarkus:add-extension -Dextensions="smallrye-reactive-messaging-kafka"
+----
+
+This will add the following to your `pom.xml`:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-smallrye-reactive-messaging-kafka</artifactId>
+</dependency>
+----
 
-The solution is located in the `kafka-quickstart` {quickstarts-tree-url}/kafka-quickstart[directory].
+== Configuring Smallrye Kafka Connector
 
-== Creating the Maven Project
+Because Smallrye Reactive Messaging framework supports different messaging backends like Apache Kafka, AMQP, Apache Camel, JMS, MQTT, etc., it employs a generic vocabulary:
 
-First, we need a new project. Create a new project with the following command:
+- Applications send and receive *messages*. A message wraps a _payload_ and can be extended with some _metadata_. With the Kafka connector, a _message_ corresponds to a Kafka _record_.
+- Messages transit on *channels*. Application components connect to channels to publish and consume messages. The Kafka connector maps _channels_ to Kafka _topics_.
+- Channels are connected to message backends using *connectors*. Connectors are configured to map incoming messages to a specific channel (consumed by the application) and collect outgoing messages sent to a specific channel. Each connector is dedicated to a specific messaging technology. For example, the connector dealing with Kafka is named `smallrye-kafka`.
 
-[source,bash,subs=attributes+]
+A minimal configuration for the Kafka connector with an incoming channel looks like the following:
+
+[source, properties]
 ----
-mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
-    -DprojectGroupId=org.acme \
-    -DprojectArtifactId=kafka-quickstart \
-    -DclassName="org.acme.kafka.PriceResource" \
-    -Dpath="/prices" \
-    -Dextensions="resteasy-reactive,smallrye-reactive-messaging-kafka"
-cd kafka-quickstart
+kafka.bootstrap.servers=kafka:9092 <1>
```

Comment:
I would remove it, or use `%prod.kafka.bootstrap.servers`, as we want to encourage the usage of dev services.